src/Secrets/Secrets.php

```diff
@@ -0,0 +1,55 @@
+<?php declare(strict_types=1);
+
+namespace Bref\Secrets;
+
+use AsyncAws\Ssm\SsmClient;
+
+class Secrets
+{
+    /**
+     * Decrypt environment variables that are encrypted with AWS SSM.
+     *
+     * @param SsmClient $ssmClient To allow mocking in tests.
+     */
+    public static function decryptSecretEnvironmentVariables(?SsmClient $ssmClient = null): void
+    {
+        /** @var array<string,string>|string|false $envVars */
+        $envVars = getenv(local_only: true); // @phpstan-ignore-line PHPStan is wrong
+        if (! is_array($envVars)) {
+            return;
+        }
+
+        // Only consider environment variables that start with "bref-ssm:"
+        $envVarsToDecrypt = array_filter($envVars, function (string $value): bool {
+            return str_starts_with($value, 'bref-ssm:');
+        });
+        if (empty($envVarsToDecrypt)) {
+            return;
+        }
+
+        // Extract the SSM parameter names by removing the "bref-ssm:" prefix
+        $ssmNames = array_map(function (string $value): string {
+            return substr($value, strlen('bref-ssm:'));
+        }, $envVarsToDecrypt);
+
+        $ssm = $ssmClient ?? new SsmClient([
+            'version' => 'latest',
+            'region' => $_ENV['AWS_REGION'] ?? $_ENV['AWS_DEFAULT_REGION'],
+        ]);
+
+        $parameters = $ssm->getParameters([
+            'Names' => array_values($ssmNames),
```

Comment:
SSM only supports fetching 10 parameters at once, so you may want to chunk these

https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_GetParameters.html

![Screen Shot 2023-01-07 at 11 33 30](https://user-images.githubusercontent.com/1010392/211163228-8172f3dc-752b-4bdb-8941-81a5a50c6d26.png)

Reply:
Thank you @chrisguitarguy, that should be fixed now!